### PR TITLE
Added timeout when sending on UDP. [5956]

### PR DESCRIPTION
**Pull request by richiware**

### CODE REVIEW

File: src/cpp/transport/UDPTransportInterface.cpp [hunk 1]
```diff
@@ -393,6 +396,25 @@ bool UDPTransportInterface::send(
...
+                timeStruct.tv_sec = 0;
+                timeStruct.tv_usec = timeout.count();
+                setsockopt(getSocketPtr(socket)->native_handle(), SOL_SOCKET, SO_SNDTIMEO,
+                        reinterpret_cast<const char*>(&timeStruct), sizeof(timeStruct));
+            }
+            else
+            {
+                struct timeval timeStruct;
+                timeStruct.tv_sec = 0;
+                timeStruct.tv_usec = 0;
```
Comment (LuisGP) [inline src/cpp/transport/UDPTransportInterface.cpp]: I think both branches can be merged with the following change:
```suggestion
                timeStruct.tv_usec = timeout.count() > 0 ? timeout.count() : 0;
```

Comment (richiware) [reply src/cpp/transport/UDPTransportInterface.cpp]: Done

**Comment by richiware** [comment]
Build status:
                                * Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/null/)
                                * Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/null/)
                                * Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/null/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/64/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/0/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/0/)

File: src/cpp/rtps/participant/RTPSParticipantImpl.cpp [hunk 2]
```diff
@@ -998,7 +998,11 @@ bool RTPSParticipantImpl::sendSync(
 
         for (auto& send_resource : send_resource_list_)
         {
-            send_resource->send(msg->buffer, msg->length, destination_loc);
+            // Calculate next timeout.
+            std::chrono::microseconds timeout =
+                std::chrono::duration_cast<std::chrono::microseconds>(max_blocking_time_point - std::chrono::steady_clock::now());
```
[the remark this answers is not in the document]
Comment (richiware) [reply src/cpp/rtps/participant/RTPSParticipantImpl.cpp]: Done

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/0/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/0/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/68/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/120/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/109/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/119/)

**Comment by raquelalvarezbanos** [comment]
Failing tests are the usual ones. Merging this in.

**Pull request merged**